hi-ml-cpath/src/health_cpath/utils/output_utils.py

```diff
@@ -205,7 +206,12 @@ def should_save_validation_outputs(self, metrics_dict: Mapping[MetricsKey, Metri
         :return: Whether this is the best validation epoch so far.
         """
         # The metric needs to be computed on all ranks to allow synchronisation
-        metric_value = float(metrics_dict[self.primary_val_metric].compute())
+        metric = metrics_dict[self.primary_val_metric]
+        if not metric._update_called:
```

Comment:
What's the reasoning here?

Reply:
Without it we see this warning: `UserWarning: The ``compute`` method of metric Accuracy was called before the ``update`` method which may lead to errors, as metric states have not yet been updated.` 
If the metric hasn't been updated then it is erroneous so we don't want to save it

Reply:
I'm a little confused here since on `validation_epoch_end` we call `log_metrics(ModelKey.VAL)` which iteratively calls `metric.log()` on every metric in the validation dictionary. This implicitly calls `.compute()` and then `.reset()` on each metric. So by the point this part of the code is reached - all validation metrics should have been reset and therefore there is nothing to compute...

Reply:
I've added a warning to catch any case like this